Index: nfs-utils-1.0.10/support/nfs/svc_socket.c
===================================================================
--- nfs-utils-1.0.10.orig/support/nfs/svc_socket.c
+++ nfs-utils-1.0.10/support/nfs/svc_socket.c
@@ -64,7 +64,7 @@ svc_socket (u_long number, int type, int
 	}
     }
 
-  __bzero ((char *) &addr, sizeof (addr));
+  memset (&addr, 0, sizeof (addr));
   addr.sin_family = AF_INET;
 
   ret = getrpcbynumber_r (number, &rpcbuf, rpcdata, sizeof rpcdata,
