Index: nfs-utils-1.1.2/utils/mount/mount.c
===================================================================
--- nfs-utils-1.1.2.orig/utils/mount/mount.c	2008-07-11 18:56:00.000000000 +0200
+++ nfs-utils-1.1.2/utils/mount/mount.c	2008-07-11 18:56:06.000000000 +0200
@@ -189,9 +189,6 @@
 	}
 	if (nfs_mount_data_version > NFS_MOUNT_VERSION)
 		nfs_mount_data_version = NFS_MOUNT_VERSION;
-	else
-		if (kernel_version > MAKE_VERSION(2, 6, 22))
-			string++;
 }
 
 static void print_one(char *spec, char *node, char *type, char *opts)
