Index: nfs-utils-1.0.10/utils/mountd/mountd.c
===================================================================
--- nfs-utils-1.0.10.orig/utils/mountd/mountd.c
+++ nfs-utils-1.0.10/utils/mountd/mountd.c
@@ -558,7 +558,7 @@ main(int argc, char **argv)
 
 	/* Parse the command line options and arguments. */
 	opterr = 0;
-	while ((c = getopt_long(argc, argv, "o:n:Fd:f:p:P:hH:N:V:vs:t:", longopts, NULL)) != EOF)
+	while ((c = getopt_long(argc, argv, "o:nFd:f:p:P:hH:N:V:vs:t:", longopts, NULL)) != EOF)
 		switch (c) {
 		case 'o':
 			descriptors = atoi(optarg);
