Index: nfs-utils-1.1.0/utils/gssd/gssd.man
===================================================================
--- nfs-utils-1.1.0.orig/utils/gssd/gssd.man
+++ nfs-utils-1.1.0/utils/gssd/gssd.man
@@ -35,7 +35,7 @@ With the \-n option, "machine credential
 for accesses by UID 0.  Instead, credentials must be obtained
 manually like all other users.  Use of this option means that
 "root" must manually obtain Kerberos credentials before
-attemtpting to mount an nfs filesystem requiring Kerberos
+attempting to mount an nfs filesystem requiring Kerberos
 authentication.
 .TP
 .B -k keytab
