Index: nfs-utils-1.1.1/support/nfs/xlog.c
===================================================================
--- nfs-utils-1.1.1.orig/support/nfs/xlog.c
+++ nfs-utils-1.1.1/support/nfs/xlog.c
@@ -133,9 +133,13 @@ xlog_enabled(int fac)
 void
 xlog_backend(int kind, const char *fmt, va_list args)
 {
+	va_list args2;
+
 	if (!(kind & (L_ALL)) && !(logging && (kind & logmask)))
 		return;
 
+	va_copy(args2, args);
+
 	if (log_syslog) {
 		switch (kind) {
 		case L_FATAL:
@@ -172,10 +176,12 @@ xlog_backend(int kind, const char *fmt, 
 		fprintf(stderr, "%s: ", log_name);
 #endif
 
-		vfprintf(stderr, fmt, args);
+		vfprintf(stderr, fmt, args2);
 		fprintf(stderr, "\n");
 	}
 
+	va_end(args2);
+
 	if (kind == L_FATAL)
 		exit(1);
 }
