Index: nfs-utils-1.1.1~git-20070709/utils/mount/mount.c
===================================================================
--- nfs-utils-1.1.1~git-20070709.orig/utils/mount/mount.c
+++ nfs-utils-1.1.1~git-20070709/utils/mount/mount.c
@@ -530,9 +530,7 @@ int main(int argc, char *argv[])
 		 * Parent exits immediately with success.  Make
 		 * sure not to free "mount_point"
 		 */
-		if (fork() > 0)
-			exit(0);
-
+		daemon(0,0);
 		mnt_err = try_mount(spec, mount_point, flags, fs_type,
 					&extra_opts, mount_opts, fake,
 					nomtab, BACKGROUND);
