Index: nfs-utils-1.1.1/utils/exportfs/exportfs.man
===================================================================
--- nfs-utils-1.1.1.orig/utils/exportfs/exportfs.man
+++ nfs-utils-1.1.1/utils/exportfs/exportfs.man
@@ -167,11 +167,13 @@ all entries listed in
 are removed from the kernel export tables, and the file is cleared. This
 effectively shuts down all NFS activity.
 .P
-To remove individual export entries, one can specify a
+To remove an export to a host, specify a
 .I host:/path
 pair. This deletes the specified entry from
 .B xtab
 and removes the corresponding kernel entry (if any).
+To remove one or more exports to several hosts, use
+.BR "exportfs -ua" .
 .P
 .\" -------------------- Dumping the Export Table -----------------
 .SS Dumping the Export Table 
