--- a/utils/mountd/cache.c	2007-08-10 01:28:21.000000000 +1000
+++ b/utils/mountd/cache.c	2007-08-10 01:28:21.000000000 +1000
@@ -36,6 +36,15 @@
 #include "blkid/blkid.h"
 #endif
 
+#ifndef major
+# define major(dev) gnu_dev_major (dev)
+#endif
+#ifndef minor
+# define minor(dev) gnu_dev_minor (dev)
+#endif
+#ifndef makedev
+# define makedev(maj, min) gnu_dev_makedev (maj, min)
+#endif
 
 enum nfsd_fsid {
 	FSID_DEV = 0,
