Index: nfs-utils-1.1.1/utils/nfsstat/nfsstat.c
===================================================================
--- nfs-utils-1.1.1.orig/utils/nfsstat/nfsstat.c	2007-11-19 21:57:59.000000000 +0100
+++ nfs-utils-1.1.1/utils/nfsstat/nfsstat.c	2007-11-19 21:58:13.000000000 +0100
@@ -207,7 +207,7 @@
 {
 	printf("Usage: %s [OPTION]...\n\
 \n\
-  -m, --mounted\t\tShow statistics on mounted NFS filesystems\n\
+  -m, --mounts\t\tShow statistics on mounted NFS filesystems\n\
   -c, --client\t\tShow NFS client statistics\n\
   -s, --server\t\tShow NFS server statistics\n\
   -2\t\t\tShow NFS version 2 statistics\n\
Index: nfs-utils-1.1.1/utils/nfsstat/nfsstat.man
===================================================================
--- nfs-utils-1.1.1.orig/utils/nfsstat/nfsstat.man	2007-11-19 21:58:16.000000000 +0100
+++ nfs-utils-1.1.1/utils/nfsstat/nfsstat.man	2007-11-19 21:58:48.000000000 +0100
@@ -35,7 +35,7 @@
 .B \-4
 Print only NFS v4 statistics. 
 .TP
-.B \-m, \-\-mounted
+.B \-m, \-\-mounts
 Print information about each of the mounted \fBNFS\fR file systems.
 
 If this option is used, all other options are ignored.
