Index: nfs-utils-1.0.10/utils/lockd/lockd.man
===================================================================
--- nfs-utils-1.0.10.orig/utils/lockd/lockd.man
+++ nfs-utils-1.0.10/utils/lockd/lockd.man
@@ -7,7 +7,7 @@
 .SH NAME
 rpc.lockd \- start kernel lockd process
 .SH SYNOPSIS
-.B "/usr/sbin/rpc.lockd
+.B "/sbin/rpc.lockd
 .SH DESCRIPTION
 The
 .B rpc.lockd
Index: nfs-utils-1.0.10/utils/showmount/showmount.man
===================================================================
--- nfs-utils-1.0.10.orig/utils/showmount/showmount.man
+++ nfs-utils-1.0.10/utils/showmount/showmount.man
@@ -4,7 +4,7 @@
 .SH NAME
 showmount \- show mount information for an NFS server
 .SH SYNOPSIS
-.B /usr/sbin/showmount
+.B /sbin/showmount
 .B "[\ \-adehv\ ]"
 .B "[\ \-\-all\ ]"
 .B "[\ \-\-directories\ ]"
