Index: nfs-utils-1.1.1/utils/mountd/cache.c
===================================================================
--- nfs-utils-1.1.1.orig/utils/mountd/cache.c
+++ nfs-utils-1.1.1/utils/mountd/cache.c
@@ -181,8 +181,6 @@ int get_uuid(char *path, char *uuid, int
 		if (cache == NULL)
 			blkid_get_cache(&cache, NULL);
 
-		blkid_probe_all_new(cache);
-
 		if (stat(path, &stb) != 0)
 			return 0;
 		devname = blkid_devno_to_devname(stb.st_dev);
