Index: nfs-utils-1.1.1/utils/gssd/krb5_util.c
===================================================================
--- nfs-utils-1.1.1.orig/utils/gssd/krb5_util.c
+++ nfs-utils-1.1.1/utils/gssd/krb5_util.c
@@ -404,7 +404,7 @@ gssd_get_single_krb5_cred(krb5_context c
 	    cache_type = "FILE";
 	snprintf(cc_name, sizeof(cc_name), "%s:%s/%s%s_%s",
 		cache_type,
-		GSSD_DEFAULT_CRED_DIR, GSSD_DEFAULT_CRED_PREFIX,
+		ccachedir, GSSD_DEFAULT_CRED_PREFIX,
 		GSSD_DEFAULT_MACHINE_CRED_SUFFIX, ple->realm);
 	ple->endtime = my_creds.times.endtime;
 	if (ple->ccname != NULL)
