From: Ben Hutchings <ben@decadent.org.uk>
Subject: [PATCH] mount.nfs.man, nfs.man: Update distinction between fstypes

This addresses Debian bug #575503.

--- nfs-utils.orig/utils/mount/mount.nfs.man
+++ nfs-utils/utils/mount/mount.nfs.man
@@ -15,16 +15,20 @@
 .BR mount (8)
 command for mounting NFS shares. This subcommand, however, can also be used as a standalone command with limited functionality.
 
-.BR mount.nfs4 
-is used for mounting NFSv4 file system, while 
-.BR mount.nfs 
-is used to mount NFS file systems versions 3 or 2.
 .I remotetarget 
 is a server share usually in the form of 
 .BR servername:/path/to/share.
 .I dir 
 is the directory on which the file system is to be mounted.
 
+Under Linux 2.6.32 and later kernel versions,
+.BR mount.nfs
+can mount all NFS file system versions.  Under earlier Linux kernel versions,
+.BR mount.nfs4
+must be used for mounting NFSv4 file systems while
+.BR mount.nfs
+must be used for NFSv3 and v2.
+
 .SH OPTIONS
 .TP
 .BI "\-r"
--- nfs-utils.orig/utils/mount/nfs.man
+++ nfs-utils/utils/mount/nfs.man
@@ -71,14 +71,9 @@
 .P
 The
 .I fstype
-field contains either "nfs" (for version 2 or version 3 NFS mounts)
-or "nfs4" (for NFS version 4 mounts).
-The
-.B nfs
-and
-.B nfs4
-file system types share similar mount options,
-which are described below.
+field usually contains "nfs". For NFSv4 file systems on Linux
+kernel versions before 2.6.32, it must contain "nfs4" (this is
+also supported on later versions).
 .SH "MOUNT OPTIONS"
 Refer to
 .BR mount (8)
@@ -89,14 +84,23 @@
 in
 .IR /etc/fstab .
 .DT
-.SS "Valid options for either the nfs or nfs4 file system type"
-These options are valid to use when mounting either
-.B nfs
-or
-.B nfs4
-file system types.
+.SS "Valid options for all NFS file system versions"
 They imply the same behavior
-and have the same default for both file system types.
+and have the same default for all file system versions.
+.TP 1.5i
+.BI nfsvers= n
+The NFS protocol version number used to contact the server's NFS service.
+If the server does not support the requested version,
+the mount request fails.
+If this option is not specified, the client attempts to use the latest
+version (4 or 3), but negotiates the NFS version with the server if this
+is not available.
+.TP 1.5i
+.BI vers= n
+This option is an alternative to the
+.B nfsvers
+option.
+It is included for compatibility with other operating systems.
 .TP 1.5i
 .BR soft " / " hard
 Determines the recovery behavior of the NFS client
@@ -476,11 +480,7 @@
 .IP
 The DATA AND METADATA COHERENCE section contains a
 detailed discussion of these trade-offs.
-.SS "Valid options for the nfs file system type"
-Use these options, along with the options in the above subsection,
-for mounting the
-.B nfs
-file system type.
+.SS "Valid options for NFSv2 and v3 file systems"
 .TP 1.5i
 .BI proto= netid
 The transport protocol name and protocol family the NFS client uses
@@ -621,23 +621,6 @@
 reports the proper maximum component length to applications
 in such cases.
 .TP 1.5i
-.BI nfsvers= n
-The NFS protocol version number used to contact the server's NFS service.
-The Linux client supports version 2 and version 3 of the NFS protocol
-when using the file system type
-.BR nfs .
-If the server does not support the requested version,
-the mount request fails.
-If this option is not specified, the client attempts to use version 3,
-but negotiates the NFS version with the server if version 3 support
-is not available.
-.TP 1.5i
-.BI vers= n
-This option is an alternative to the
-.B nfsvers
-option.
-It is included for compatibility with other operating systems.
-.TP 1.5i
 .BR lock " / " nolock
 Selects whether to use the NLM sideband protocol to lock files on the server.
 If neither option is specified (or if
@@ -727,11 +710,7 @@
 on NFS version 3 mounts to read small directories.
 Some applications perform better if the client uses only READDIR requests
 for all directories.
-.SS "Valid options for the nfs4 file system type"
-Use these options, along with the options in the first subsection above,
-for mounting the
-.B nfs4
-file system type.
+.SS "Valid options for NFSv4 file systems"
 .TP 1.5i
 .BI proto= netid
 The transport protocol name and protocol family the NFS client uses
@@ -851,12 +830,11 @@
 To mount using NFS version 4,
 use the
 .B nfs4
-file system type.
-The
-.B nfsvers
-mount option is not supported for the
-.B nfs4
-file system type.
+file system type or (from Linux 2.6.32) use the
+.B nfs
+file system type and specify the
+.B nfsvers=4
+mount option.
 .P
 The following example from an
 .I /etc/fstab
