Description: Adjust program paths in systemd units
Author: Martin Pitt <martin.pitt@ubuntu.com>
Forwarded: No

--- a/systemd/rpc-statd-notify.service	2021-06-11 04:07:47.000000000 +1000
+++ b/systemd/rpc-statd-notify.service	2021-09-05 18:26:31.577981201 +1000
@@ -12,5 +12,5 @@ PartOf=nfs-utils.service
 
 [Service]
 Type=forking
-ExecStart=-/usr/sbin/sm-notify
+ExecStart=-/sbin/sm-notify
 RemainAfterExit=yes
--- a/systemd/rpc-statd.service	2021-06-11 04:07:47.000000000 +1000
+++ b/systemd/rpc-statd.service	2021-09-05 18:32:16.682315831 +1000
@@ -14,4 +14,4 @@ IgnoreOnIsolate=yes
 Environment=RPC_STATD_NO_NOTIFY=1
 Type=forking
 PIDFile=/run/rpc.statd.pid
-ExecStart=/usr/sbin/rpc.statd
+ExecStart=/sbin/rpc.statd
