From: James Youngman <james@youngman.org>
Date: Mon, 25 Sep 2023 08:53:32 -0400
Subject: Remove extraneous words left behind by commit 522837f.
Origin: https://git.linux-nfs.org/?p=steved/nfs-utils.git;a=commit;h=fb635d63fa7d78d5a80c9b638d177c5a78b64124
Bug-Debian: https://bugs.debian.org/1051088

Signed-off-by: Steve Dickson <steved@redhat.com>
---
 utils/mount/nfs.man | 1 -
 1 file changed, 1 deletion(-)

diff --git a/utils/mount/nfs.man b/utils/mount/nfs.man
index 9b4bc9c9cd4f..c0ba4d0845ac 100644
--- a/utils/mount/nfs.man
+++ b/utils/mount/nfs.man
@@ -1013,7 +1013,6 @@ file. See
 .BR nfsmount.conf(5)
 for details.
 .SH EXAMPLES
-mount option.
 To mount using NFS version 3,
 use the
 .B nfs
-- 
2.40.1

